controlpanel/frontend/views/dashboard.py

```diff
@@ -0,0 +1,303 @@
+    model = Dashboard
+    permission_required = "api.register_dashboard"
+    template_name = "dashboard-register.html"
+
+    def get_success_url(self):
+        messages.success(
+            self.request,
+            f"Successfully registered {self.object.name} dashboard",
+        )
+        return reverse_lazy("manage-dashboard", kwargs={"pk": self.object.pk})
+
+    def form_valid(self, form):
+        # add dashboard, set creator as an admin and viewer
+        with transaction.atomic():
+            user = self.request.user
+            email = user.justice_email.lower()
+            dashboard = form.save()
+            dashboard.created_by = user
+            dashboard.admins.add(user)
+            viewer, created = DashboardViewer.objects.get_or_create(email=email)
+            dashboard.viewers.add(viewer)
+            self.object = dashboard
+            return HttpResponseRedirect(self.get_success_url())
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class DashboardDetail(OIDCLoginRequiredMixin, PermissionRequiredMixin, DetailView):
+    context_object_name = "dashboard"
+    model = Dashboard
+    permission_required = "api.retrieve_dashboard"
+    template_name = "dashboard-detail.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        dashboard = self.get_object()
+        dashboard_admins = dashboard.admins.all()
+
+        context["admin_options"] = User.objects.exclude(
+            auth0_id="",
+        ).exclude(
+            auth0_id__in=[user.auth0_id for user in dashboard_admins],
+        )
+
+        context["dashboard_admins"] = dashboard_admins
+
+        context["grant_access_form"] = GrantDomainAccessForm(
+            dashboard=dashboard,
+        )
+
+        return context
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class DeleteDashboard(OIDCLoginRequiredMixin, PermissionRequiredMixin, DeleteView):
+    model = Dashboard
+    permission_required = "api.destroy_dashboard"
+    success_url = reverse_lazy("list-dashboards")
+    allowed_methods = ["POST"]
+
+    def form_valid(self, form):
+        dashboard = self.get_object()
+        dashboard.delete()
+        messages.success(self.request, f"Successfully deleted {dashboard.name} dashboard")
+        return HttpResponseRedirect(self.get_success_url())
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class UpdateDashboard(
+    OIDCLoginRequiredMixin,
+    PermissionRequiredMixin,
+    SingleObjectMixin,
+    RedirectView,
+):
+    http_method_names = ["post"]
+    model = Dashboard
+
+    def get_redirect_url(self, *args, **kwargs):
+        return reverse_lazy("manage-dashboard", kwargs={"pk": kwargs["pk"]})
+
+    def post(self, request, *args, **kwargs):
+        self.perform_update(**kwargs)
+        return super().post(request, *args, **kwargs)
+
```

Comment:
To make it clear that subclasses need to implement `perform_update` I would add:
```
def perform_update(self):
    raise NotImplementedError("Subclasses must define this method")
```